test/functional/master/test_deallocation_and_allocation_of_slaves_mid_build.py

```diff
@@ -1,11 +1,14 @@
 import os
 import tempfile
+from unittest import skipIf
 import yaml
 
+from app.util.process_utils import is_windows
 from test.framework.functional.base_functional_test_case import BaseFunctionalTestCase
 from test.functional.job_configs import JOB_WITH_SLEEPS
 
 
+@skipIf(is_windows(), 'Fails on AppVeyor; see issue #345')
```

Comment:
Can we just increase the timeout for master.block_until_build_finished() to 20 seconds or something instead of skipping this on Windows?